src/main/java/dataStructures/hashSet/openAddressing/HashSet.java

```diff
@@ -99,19 +103,28 @@ public boolean add(T element) {
      */
     public boolean remove(T element) {
         // If load factor falls below 0.25 and still above minimum size (16), shrink the hashset by half.
-        if (this.size() <= this.capacity() * 0.25 && this.capacity() / 2 >= INITIAL_CAPACITY) {
+        if (size() <= capacity() * 0.25 && capacity() / 2 >= INITIAL_CAPACITY) {
             resize(this.capacity() / 2);
         }
 
-        int bucketIndex = this.search(element);
-        // If the index returned by the probe function contains an empty bucket, then the element is not present in
-        // the set.
-        if (bucketIndex == ELEMENT_NOT_FOUND) {
-            return false;
+        int collisions = 0;
+        while (collisions < capacity()) {
+            int bucketIndex = hashFunction(element, collisions);
+
+            // Element is not removed, because it is not in the Set.
+            if (isNullBucket(bucketIndex)) {
```

Comment:
here is specifically null check right? if it holds a tombstone, ignore and move on right? could you add a in-line comment to remind readers and clear any possible cause of confusion

Reply:
Resolved in a1d46976c1168ff2b597453fa8d7876f3dc57d7b